code/modules/mining/aux_base.dm

```diff
@@ -237,6 +241,22 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/computer/auxiliary_base, 32)
 	to_chat(user, span_notice("Landing zone set."))
 	return ZONE_SET
 
+/*
+ * Add [new_turret] to our list of turrets, and handle the case in which they're pre-emptively deleted.
+ */
+/obj/machinery/computer/auxiliary_base/proc/add_turret(obj/machinery/porta_turret/aux_base/new_turret)
+	LAZYADD(turrets, new_turret)
+	RegisterSignal(new_turret, COMSIG_PARENT_QDELETING, .proc/remove_turret)
```

Comment:
I try to avoid the qdeleting signal because of the potential for conflicts (past me was not so wise)
Is there any reason turrets can't be weakrefs?

Reply:
The aux base UI iterates over the turrets list in `ui_data` and `ui_act` a few times, and I wasn't sure how nice it would be to have to resolve every weakref in the list every ui update

Reply:
that'd be acceptable, specially since people don't typically open that ui a lot. It's not a huge deal regardless, resolve() isn't horrid

Reply:
It doesn't *feel* as clean with weakrefs but oh well